src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.NavigationalCommandTarget.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal sealed partial class DifferenceViewerPreview
+    {
+        private sealed class NavigationalCommandTarget(ITextView textView, IEditorOperations editorOperations) : IOleCommandTarget
```

Comment:
All the code in this file is a port from https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/496210

Reply:
i really hate this.  is there any plan to make a reusable component here?  this seems to be taking us backwards wrt to being able to use the editor cleanly.